### DISCUSSION TITLE
Translator and JSON login

### DISCUSSION THREAD

**Discussion by marcelowzd** [symfony/symfony#58446] [Q&A]
Hello people,

So, before i open it as a feature request or issue, i would like to understand if it is what i understood from it.

I'm currently using Symfony 7.1 and json_login in a new project and if i enter the wrong credentials, i get the "Invalid credentials" message and a 401 code.
I saw inside the JsonLoginAuthenticator that in this message it calls the translator, but there are many others where it calls only the sprintf. I would like to know if this is on purpose or maybe i didn't see something i should have (there are other authenticators doing the same thing).

Thanks in advance.

**Comment by MatTheCat** [comment]
Hello,

I’m not sure what are those `sprintf` calls you mention, but the difference is probably who is supposed to read the message: an exception’s will be for developers so it does not need to be translated, whereas the `JsonLoginAuthenticator`’s is part of an HTTP response so it may need to be translated if the client is displaying it verbatim to its user.
